src/routes/ContactListRoute/ContactListRoute.tsx

```diff
@@ -30,13 +33,60 @@ export const ContactListRoute = () => {
     return -1
   }).filter(contact => contact.name.toLowerCase().includes(search.toLowerCase()))
 
+  const handleDownloadContacts = async() => {
+    if (!sortedContacts) return
+    setIsDownloading(true)
+
+    const limit = RateLimit(5) // requests per second
```

Comment:
limit rate request to a max of 5 per second. This is air table's current max via the api